solidity/test/rewards/TestECDSABackportRewards.js

```diff
@@ -0,0 +1,254 @@
+const BondedECDSAKeep = contract.fromArtifact("BondedECDSAKeep")
+const ECDSABackportRewards = contract.fromArtifact("ECDSABackportRewards")
+
+const KeepTokenGrant = contract.fromArtifact("TokenGrant")
+
+const BN = web3.utils.BN
+
+const chai = require("chai")
+chai.use(require("bn-chai")(BN))
+const expect = chai.expect
+
+describe("ECDSABackportRewards", () => {
+  let registry
+  let rewardsContract
+  let keepToken
+
+  let tokenStaking
+  let tokenGrant
+  let keepFactory
+  let bondedSortitionPoolFactory
+  let keepBonding
+  let randomBeacon
+  let keepMembers
+
+  const owner = accounts[0]
+
+  const keepSize = 3
+  const numberOfCreatedKeeps = 41
+  const tokenDecimalMultiplier = web3.utils.toBN(10).pow(web3.utils.toBN(18))
+
+  // 1,000,000,000 - total KEEP supply
+  //   200,000,000 - 20% of the total supply goes to staker rewards
+  //   180,000,000 - 90% of staker rewards goes to the ECDSA stakers
+  //     1,800,000 - 1% of ECDSA staker rewards goes to May - Sep keeps
+  const ECDSABackportKEEPRewards = web3.utils
+    .toBN(1800000)
+    .mul(tokenDecimalMultiplier)
+
+  let keepCreationTimestamp = 1589408353
+
+  before(async () => {
+    await BondedSortitionPoolFactory.detectNetwork()
+    await BondedSortitionPoolFactory.link(
+      "StackLib",
+      (await StackLib.new({from: owner})).address
+    )
+
+    await initializeNewFactory()
+    keepMembers = await createMembers()
+    rewardsContract = await ECDSABackportRewards.new(
+      keepToken.address,
+      keepFactory.address,
+      {from: owner}
+    )
+
+    await fund(ECDSABackportKEEPRewards)
+    await createKeeps()
+  })
+
+  beforeEach(async () => {
+    await createSnapshot()
+  })
+
+  afterEach(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("interval allocation", async () => {
+    it("should equal the full allocation", async () => {
+      const expectedAllocation = 1800000
+
+      await rewardsContract.allocateRewards(0)
+
+      const allocated = (await rewardsContract.getAllocatedRewards(0)).div(
+        tokenDecimalMultiplier
+      )
+      expect(allocated).to.eq.BN(expectedAllocation)
+    })
+  })
+
+  describe("rewards withdrawal", async () => {
+    it("should correctly distribute rewards between beneficiaries", async () => {
+      // 1800000 / 41 / 3 = 14634.14634 KEEP.
+      // Each beneficiary receive 14634 KEEP.
+      // Decimals (0.14634) are rolled over to the last keep signer.
+      //
+      // All 3 signers belong to all 41 keeps for testing purposes.
+      // KEEP is added to the signers in every iteration; total 41 times (number of keeps).
+      const expectedSingleReward = new BN(14634)
+
+      for (let i = 0; i < numberOfCreatedKeeps; i++) {
+        const keepAddress = await keepFactory.getKeepAtIndex(i)
+        await rewardsContract.receiveReward(keepAddress)
+
+        await assertKeepBalanceOfBeneficiaries(i, expectedSingleReward)
+      }
+    })
```

Comment:
I'd simplify it to checking just the final balances. The logic inside `assertKeepBalanceOfBeneficiaries` with `keepNumber` is harder to grasp and individual distributions are tested in unit tests for parent `Rewards` contract anyway. Finally, we know that in practice we are talking about 3 operators for this backport. 🤷 

For example:
```
    it.only("should correctly distribute rewards between beneficiaries", async () => {
      // 1800000 / 3 = 600000
      const expectedBeneficiaryBalance = web3.utils.toBN(600000)

      for (let i = 0; i < numberOfCreatedKeeps; i++) {
        const keepAddress = await keepFactory.getKeepAtIndex(i)
        await rewardsContract.receiveReward(keepAddress)
      }

      assertKeepBalanceOfBeneficiaries(expectedBeneficiaryBalance) 
    })

    async function assertKeepBalanceOfBeneficiaries(expectedBalance) {
      // solidity is not very good when it comes to floating point precision,
      // we are allowing for +-1 KEEP difference margin between expected and
      // actual value
      const precision = 1

      for (let i = 0; i < keepMembers.length; i++) {
          const balance = await keepToken.balanceOf(keepMembers[i].beneficiary)
          const balanceInKeep = balance.div(tokenDecimalMultiplier)
          expect(balanceInKeep).to.gte.BN(expectedBalance.subn(precision))
          expect(balanceInKeep).to.lte.BN(expectedBalance.addn(precision))
      }
  }
```

Reply:
Ok, done.